src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -459,6 +459,28 @@ void primitive_inst::update_shape() {
     }
 }
 
+kernel_impl_params primitive_inst::get_fake_aligned_params_if_possible(kernel_impl_params const& orig_impl_param) {
+    auto updated_params = _node->type()->get_fake_aligned_params(orig_impl_param);
+
+    const auto &dev_info = get_node().get_program().get_engine().get_device_info();
+
+    // The target HW of this patch is limited because of performance concern
+    if (dev_info.supports_immad && dev_info.dev_type == device_type::integrated_gpu) {
+        // Check whether the input node has enough space for output data. Otherwise, fake alignment is not possible due to page fault
+        // i.e. predecessor node was supposed be increased already
+        if (get_node().is_type<fully_connected>() && dependencies().size() > 0 && dep_memory(0).get_layout().is_static()
```

Comment:
Can it be a part of fully_connected_inst::get_fake_aligned_params ? There are few checks on fake alignment applicability, so why do we need to move these checks outside?

Reply:
Currently it is declared as static function. I think we can apply this within the existing get_fake_aligned_params if we fix this as just a normal member function..

Reply:
LGTM, lets consider in next PR

Reply:
Agreed with Vladimir P. that this comment should be addressed in the next PR. This PR can be merged to have green light to run nightly validation.